### DISCUSSION TITLE
Using pool with S4 class types from dpm package output

### DISCUSSION THREAD

**Discussion by sccollitt** [amices/mice#400] [Impute, analyse and pool]
Hi,
I'm looking to use the [dpm package](https://github.com/jacob-long/dpm) to model data, using a list of multiply imputed dataframes.  However, the list of objects resulting from the "dpm" function are of type "S4(dpm;;dpm)."  When I attempt to use mice's "pool" function on this list, I receive the following error:

"Error: $ operator not defined for this S4 class"

Do you know if there's a way to pool these model estimates using mice's pool function?
Thank you for your time and for your work in creating and maintaining the mice package.  It's helped me immensely in my research.

Sam

**Comment by stefvanbuuren** [comment]
Sam, 

This issue is better placed as a discussion, so I am transferring. 

The dpm package has `tidy` and `glance` tidiers, so please check whether you can extract the relevant model components.

**Reply by sccollitt** [reply]
Hi Stef, would this just be the coefs and vcovs?  Or point estimates, standard errors, and model residual degrees of freedom?  Once I have these components, what can I do with them with pool()?  Thanks for your time.  I have also reached out to Jacob Long on this from the dpm side.

Sam
